lib/zendesk-uploader/api-utils.js

```diff
@@ -4,28 +4,27 @@ const constants = require('./constants');
 
 /**
  * Creates and returns a zendesk API client
+ *
+ * @param {Config} config - global docpub configuraion
+ *
  * @returns {Object} instance of the created zendesk API client
- * @throws {Error} Will throw an error if the environment variables for the api client are not set
+ * @throws {Error} Will throw an error if config is missing
  */
-const getClient = function() {
-    if (!process.env.ZENDESK_API_USERNAME) {
-        throw (new Error('Environment variable for Zendesk Username is undefined'));
+const getClient = function(config) {
+    if (!config) {
+        throw new Error('Unable to configure ZenDesk client without a config');
```

Comment:
This error isn't being handled anywhere, so it will have ugly output if run without a config file.

It needs to be caught somewhere and output with the logger. Although I'm not sure where...

Maybe the config file shouldn't be read until `docpub.uploadCategory()` is called?